### PR TITLE
Typo fixes

### PR DESCRIPTION
**Pull request by jamespooley**
Other than the typos, one justification and one comment:

1. The trailing comma after the parameter_tabs object is inserted into the full UI doesn't break anything when running the app, but I don't see too many trailing commas so I removed it.
2. I don't remember seeing the op-null-default %||% introduced earlier in the book and I didn't add any explanatory text about it, but assuming I didn't miss its earlier introduction it may be worth a short explanation unless you consider it common knowledge for the book's target audience.

### CODE REVIEW

File: action-dynamic.Rmd [hunk 1]
```diff
@@ -416,7 +416,7 @@ ui <- fluidPage(
         choices = c("normal", "uniform", "exponential")
       ),
       numericInput("n", "Number of samples", value = 100),
-      parameter_tabs,
+      parameter_tabs
```
Comment (hadley) [inline action-dynamic.Rmd:419]: Please leave this; it makes it much easier to later rearrange the components.

Comment (jamespooley) [reply action-dynamic.Rmd:419]: Noted, will update accordingly.

Do you have a preference for how you'd like these updates handled? For example:

* Adding another commit with trailing comma left as-is and then leaving a comment in the text for the `%||%` explanation in the `jamespooley:patch-16` for later squash into single commit when merged; vs.
* Closing this PR and starting from scratch

**Comment by hadley** [comment]
Adding on to PR is fine. You're also welcome to do multiple chapters in one PR.

File: action-dynamic.Rmd [hunk 2]
```diff
@@ -654,7 +654,7 @@ server <- function(input, output, session) {
   })
   
   output$plot <- renderPlot({
-    cols <- map_chr(col_names(), ~ input[[.x]])
+    cols <- map_chr(col_names(), ~ input[[.x]] %||% "")
```
Comment (hadley) [inline action-dynamic.Rmd:657]: I think it would be better to use `NA` here, so you could skip the next line.

Comment (hadley) [reply action-dynamic.Rmd:657]: And yeah, you can drop from the `textInput()` generation since a `NULL` is an ok value for `value`.

Comment (jamespooley) [reply action-dynamic.Rmd:657]: Replacing the ` %||% ""` with `%||% NA` in the `map_chr()` anonymous function results in `Warning: Error in rect: invalid color name ''` if you drop the `cols[cols == ""] <- NA` line, and the palette plot will only show up when you've entered the correct number of (valid) color names.

Both ` %||% NA` or `%||%""` do, however, get rid of the `Warning: Error in : Result 1 must be a single string, not NULL of length 0`. They both just need to handle the `""` to `NA` conversion. 

If going the tidyverse route, something like

`cols <- map_chr(col_names(), ~ input[[.x]] %||% NA) %>% na_if("")`

would work, but I'm not convinced this is better, for the sake of clarity, than your example with `cols[cols == ""] <- NA` on its own line.

Comment (hadley) [reply action-dynamic.Rmd:657]: Having look at this in more detail, I think a better approach is to construct a `default_val()` function specifically for shiny and use that. Do you mind reverting your changes here, and then I'll fix locally.

Comment (jamespooley) [reply action-dynamic.Rmd]: Reverted the code changes, but kept the prose typos I'd found

**Comment by hadley** [comment]
Thanks!

**Pull request merged**